roles/srv-announcer/tasks/start.yml

```diff
@@ -0,0 +1,24 @@
+- name: starting service(s) - {{ service_name }}
+  systemd:
+    name: "{{ service_name }}.{{ item.item.key }}"
+    enabled: yes
+    state: >-
+      {{
+        'restarted' if
+          (binary_srva is defined and binary_srva.changed)
+          or
+          (item.changed is defined and item.changed)
+        else 'started'
+      }}
+    daemon_reload: "{{ 'yes' if item.changed is defined and item.changed else 'no' }}"
+  when: (item.item.value.deprecated | default(False)) == False
```

Comment:
This whole `deprecated` thing seems to be a new concept, or do we use the same approach elsewhere already?

I'd personally just have stopped the announcer service (via `systemctl stop`) every time we do maintainance, some hours before decomissioning the server entirely - Do we really want to track intermediate server states during decomissioning in version control?

Reply:
I'm not sure if we have the same use case in mind. Could you elaborate a bit, please? Also, what happens (or what would you like to happen) when running `systemctl stop`?

Reply:
If we want to stop the announcer to stop getting new connections, just stop the service.

If it's too inconvenient to do it manually, provide a playbook just stopping this (which can be limited to certain hosts).

If we want to radically change naming of these services, in ansible world we're probably better off by entirely redeploying the machine, instead of dealing with error-prone multi-staged removal phases.

Reply:
There are probably two pieces to it.

>  [...] just stop the service.

That is correct as far as I understand the tool. The "health check" needs to fail in order to make the announcer remove the entry. When the announcer itself is stopped, it wont remove the entry in the SRV record list it is responsible for - at least up until now.

The operational task to start/stop the service, that the announcer is responsible for, is [not yet implemented](https://github.com/wireapp/ansible-sft/blob/develop/roles/sft-server/tasks/start.yml#L5), but would probably be helpful due to the current announcer behaviour. What do you think? (you already proposed the playbook, I think it could also just be a flag).
Another way would be to extend the `srv-announcer` so that it would depublish the list item when being stopped (or expose a one-off sub-command that can be hooked into ` ExecStop=`).

Furthermore, the role currently assumes that multiple announcer instances (one for each SRV record <--> list item, e.g. FQDN) run on one machine. The reasoning was to let the announcer also maintain the the metrics SRV record. But, since we decided to maintain that one In Terraform, the overall approach might not longer be valid. Anyway, the [role API](https://github.com/wireapp/ansible-sft/blob/add-srv-announcer-role/roles/srv-announcer/defaults/main.yml#L40) for this could still be useful, no?

The `deprecated` flag is to facilitate add/removal `srv-announcer` instances on a machine over time. If neither multiple instances on one machine nor add/removal over time are valid use cases, the `loop`s for the unit file and service tasks could be removed. To facility the use case of managing multiple SRV record <--> FQDN pairs on one machine -if desired - the `srv-announcer` could be extended to take in more than just one pair.